analyticskit/src/main/java/com/busybusy/analyticskit_android/AnalyticsEvent.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2016 - 2022 busybusy, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.busybusy.analyticskit_android
+
+import androidx.annotation.NonNull
+import java.io.Serializable
+
+/**
+ * Defines information that is needed to distribute the event to the registered analytics providers.
+ *
+ * @author John Hunt on 3/5/16.
+ */
+open class AnalyticsEvent(
```

Comment:
I really wanted this to be a `data class` but since we have `ContentViewEvent` and `ErrorEvent` that extend `AnalyticsEvent` I chose to make this an `open class` to preserve functionality of the two event sub-classes.

Reply:
sealed interface named something like `AnalyticsKitEvent` so you can get away from having implementation inheritance?

Reply:
You're right something like the sealed interface you propose would be much cleaner. The goal for this version is to minimize breakage of the API from the perspective of the library consumer.

We'll do something like this in future versions of the library.